src/db.rs

```diff
@@ -266,6 +270,29 @@ impl DbInner {
         }
     }
 
+    pub async fn write_with_options(
```

Comment:
I considered updating all put/delete operations to call `write_with_options`. This unifies all writes into a single path (the commit pipeline in `batch_write.rs`).

I opted to leave the change out of this PR, though, since the new pipeline returns a `Result<(), SlateDBError>`, which ends up getting propagated to all of the put/get operations. This is _good_ (and required for https://github.com/slatedb/slatedb/issues/111), but a bigger change since it affects public APIs. Leaving it for a separate PR.

Reply:
What if we create a `WriteBatch` for the user inside the `put/delete` function?

Reply:
That's what I did. It works great. But we then have to either panic on error or continue propagating it towards. Totally doable, but I wanna keep it as a separate pull request.